### PR TITLE
 Rework Connection to DB to support non-SQL connections

### PR DESCRIPTION
**Pull request by pfu3tz**
In order to achieve this goal, multiple core classes had to change the generics to add a new one C. The connection object of the sql package is now packed into the class SQLConnection. There is now a SQLGlobalState that infers the SQLConnection for the global state and the SQLProviderAdapter infers this for the DBMS specific providers.

### CODE REVIEW

File: src/sqlancer/StatementExecutor.java [hunk 1]
```diff
@@ -59,6 +60,7 @@ public void executeStatements() throws Exception {
             assert nextAction != null;
             assert nrRemaining[i] > 0;
             nrRemaining[i]--;
+            @SuppressWarnings("rawtypes")
```
Comment (mrigger) [inline src/sqlancer/StatementExecutor.java:63]: Would `Query<?>` work here, rather than suppressing the warning?

**Review by mrigger — COMMENTED** [review]
LGTM, thanks! I have a minor improvement suggestion. Also, for the next PRs, please pay attention to the commit guidelines (specifically the format of commit messages and using `git rebase` rather than merges), see https://github.com/sqlancer/sqlancer/blob/master/CONTRIBUTING.md#commit-history.

File: src/sqlancer/StatementExecutor.java (hunk 1 above)
Comment (pfu3tz) [reply src/sqlancer/StatementExecutor.java:63]: I acually tried that before and it gives a type exception with line 69 executeStatement().

Comment (pfu3tz) [reply src/sqlancer/StatementExecutor.java:63]: I think the problem comes from the fact that StatementExecuter does not hold the generics for connection. But as soon as we give StatementExecuter this type extension we probably get into trouble again at other locations...

**Pull request merged**